src/VCAuthn/IdentityServer/Endpoints/AuthorizationEndpoint/AuthorizeEndpoint.cs

```diff
@@ -0,0 +1,211 @@
+            }
+            
+            if (!context.Request.HasFormContentType)
+            {
+                return new StatusCodeResult(HttpStatusCode.UnsupportedMediaType);
+            }
+                
+            var values = context.Request.Form.AsNameValueCollection();
+            
+            var clientResult = await _clientValidator.ValidateAsync(context);
+            if (clientResult.Client == null)
+            {
+                return Error(OidcConstants.TokenErrors.InvalidClient);
+            }
+
+            var scopes = values.Get(IdentityConstants.ScopeParamName).Split(' ');
+            if (!scopes.Contains(IdentityConstants.VCAuthnScopeName))
+            {
+                return Error(IdentityConstants.MissingVCAuthnScopeError, IdentityConstants.MissingVCAuthnScopeDesc);
+            }
+            
+            var presentationRecordId = values.Get(IdentityConstants.PresentationRequestConfigIDParamName);
+            if (string.IsNullOrEmpty(presentationRecordId))
+            {
+                return Error(IdentityConstants.InvalidPresentationRequestConfigIDError, IdentityConstants.InvalidPresentationRequestConfigIDDesc);
+            }
+            
+            var redirectUrl = values.Get(IdentityConstants.RedirectUriParameterName);
+            if (string.IsNullOrEmpty(redirectUrl))
+            {
+                return Error(IdentityConstants.InvalidRedirectUriError);
+            }
+            
+            if (clientResult.Client.RedirectUris.Any() && !clientResult.Client.RedirectUris.Contains(redirectUrl))
+            {
+                return Error(IdentityConstants.InvalidRedirectUriError);
+            }
+            
+            var responseType = values.Get(IdentityConstants.ResponseTypeUriParameterName);
+            if (string.IsNullOrEmpty(responseType))
+            {
+                responseType = IdentityConstants.DefaultResponseType;
+            }
+
+            var responseMode = values.Get(IdentityConstants.ResponseModeUriParameterName);
+            if (string.IsNullOrEmpty(responseMode))
+            {
+                responseMode = IdentityConstants.DefaultResponseMode;
+            }
+
+            PresentationRecord presentationRecord;
+            try
+            {
+                presentationRecord = await _presentationConfigurationService.Find(presentationRecordId);
+            }
+            catch (Exception e)
+            {
+                return Error(IdentityConstants.UnknownPresentationRecordId, "Cannot find respective record id");
+            }
+
+            var presentationRequest = BuildPresentationRequest(presentationRecord);
+            
+            // create a full and short url versions of a presentation requests
+            string shortUrl;
+            try
+            {
+                var url = string.Format("{0}?m={1}&r_uri={2}", _options.PublicOrigin, presentationRequest.ToJson().ToBase64(), redirectUrl);
```

Comment:
This redirect uri should be the shorten url linking to the verifiable credential presentation request

Reply:
taking out `r_uri` until/if we replace `http` with `didcomm` protocol